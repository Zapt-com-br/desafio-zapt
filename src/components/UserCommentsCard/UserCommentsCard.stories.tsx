import React from 'react';

import { Story, Meta } from '@storybook/react/types-6-0';
import UserCommentsCard, { UserCommentsCardProps } from '.';

export default {
  title: 'User Comments/UserCommentsCard',
  component: UserCommentsCard,
} as Meta;

export const UserCommentsCardDefault: Story<UserCommentsCardProps> = args => (
  <div
    style={{
      maxWidth: '58rem',
      height: ' 40rem',
    }}
  >
    <UserCommentsCard {...args} />
  </div>
);

UserCommentsCardDefault.args = {
  username: '@LuisCarlos',
  userSubtext: 'Luis carlos nascimento bastos',
  image:
    'https://www.pixsy.com/wp-content/uploads/2021/04/ben-sweet-2LowviVHZ-E-unsplash-1.jpeg',
  comment:
    '”Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal”',

  backgroundImage:
    'https://images.unsplash.com/photo-1626181181323-d0834fb2e2fd?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=634&q=80',
};
