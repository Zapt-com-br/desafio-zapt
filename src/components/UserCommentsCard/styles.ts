import styled, { css } from 'styled-components';
import { UserCommentsCardProps } from '.';

type WrapperProps = Pick<UserCommentsCardProps, 'backgroundImage'>;

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, backgroundImage }) => css`
    width: 100%;
    height: 40rem;
    background: url(${backgroundImage}) no-repeat;
    display: flex;
    justify-content: flex-start;
    align-items: flex-end;
    border-radius: ${theme.border.radius};
    position: relative;
    overflow: hidden;
  `}
`;

export const SocialCardWrapper = styled.div`
  z-index: 1;
`;

export const Overlay = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
`;
