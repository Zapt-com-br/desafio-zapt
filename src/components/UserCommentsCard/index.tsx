import React from 'react';

import { SocialCard } from 'components';

import * as Styled from './styles';

export type UserCommentsCardProps = {
  backgroundImage: string;
  comment: string;
  username: string;
  image: string;
  userSubtext?: string;
};

const UserCommentsCard = ({
  backgroundImage,
  comment,
  username,
  image,
  userSubtext = '',
}: UserCommentsCardProps) => (
  <Styled.Wrapper backgroundImage={backgroundImage}>
    <Styled.Overlay aria-label="overlay" />
    <Styled.SocialCardWrapper>
      <SocialCard
        maxSizeCard
        transparent
        textColor="white"
        colorUserProfile="white"
        usernameSize="medium"
        commentSize="xmedium"
        comment={comment}
        username={username}
        imageProfile={image}
        userSubtext={userSubtext}
      />
    </Styled.SocialCardWrapper>
  </Styled.Wrapper>
);

export default UserCommentsCard;
