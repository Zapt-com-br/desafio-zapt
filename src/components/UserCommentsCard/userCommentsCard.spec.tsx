import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithTheme } from 'helpers/renderWithThame';

import UserCommentsCard, { UserCommentsCardProps } from '.';

jest.mock('components/SocialCard', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="socialCard"></div>;
    },
  };
});

const props: UserCommentsCardProps = {
  backgroundImage: 'test img',
  comment: 'test comment',
  image: 'test img',
  username: 'test username',
};

describe('<UserCommentsCard />', () => {
  it('should render UserCommentsCard', () => {
    const { container } = renderWithTheme(<UserCommentsCard {...props} />);

    expect(container.firstChild).toBeInTheDocument();
    expect(screen.getByTestId('socialCard')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render UserCommentsCard whith background image correctly', () => {
    const { container } = renderWithTheme(<UserCommentsCard {...props} />);

    expect(container.firstChild).toHaveStyle({
      background: `${props.backgroundImage} no-repeat`,
    });
  });

  it('should render UserCommentsCard whith background image correctly', () => {
    renderWithTheme(<UserCommentsCard {...props} />);

    const element = screen.getByLabelText(/overlay/i);

    expect(element).toBeInTheDocument();
  });
});
