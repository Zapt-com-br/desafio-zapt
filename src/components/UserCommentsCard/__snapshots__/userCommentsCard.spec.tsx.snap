// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCommentsCard /> should render UserCommentsCard 1`] = `
.c0 {
  width: 100%;
  height: 40rem;
  background: url(test img) no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  border-radius: 0.4rem;
  position: relative;
  overflow: hidden;
}

.c2 {
  z-index: 1;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0,0,0,0.5);
}

<div
  class="c0"
>
  <div
    aria-label="overlay"
    class="c1"
  />
  <div
    class="c2"
  >
    <div
      data-testid="socialCard"
    />
  </div>
</div>
`;
